### ISSUE TITLE
Cut Bar is Slow

### ISSUE THREAD

**Issue by aolivier23** [aolivier23/edepViewer#2]
The Cut Bar GUI that each Scene gets can be very slow when processing lots of tree entries.  Cutting on properties of trajectory points and hit segments has been particularly problematic.

**Comment by aolivier23** [comment]
It occurred to me that the GPU would be great at applying cuts to OpenGL objects in parallel.  Thinking about ways to send properties of objects to the GPU in the far future.  I could extend that approach to standardize the syntax the Cut Bar uses by adding a user-specified line to the vertex shader for a Scene.  For now, a review of the code in CutBar might help.

**Comment by aolivier23** [comment]
Realized that I want to minimize the number of times I use CutBar.  Combining this with catching CutBar exceptions the first time there is a problem and ignoring CutBar from there has all but prevented CutBar from slowing down my application.  I still want to rewrite it one day to use some common language and run more quickly, but this is no longer a priority.

**Comment by aolivier23** [comment]
In the multitude of upgrades in v3.0.0, I rewrote Scene to only call UserCut when the user changes the current cut.  So, the cut bar shouldn't slow down more than one frame now, and the delay is no longer noticeable on my test machine.  If you experience problems with the cut bar, please open a new issue.
